develop/ThirdParty/Grafana/Grafana_plugin.md

```diff
@@ -0,0 +1,131 @@
+
+   - HTTP: Specify the URL to connect to your DataMiner Agent, for example `https://mydma.company.com`.
+
+    > [!CAUTION]
+    > Use a secure HTTPS connection. Without an HTTPS connection, your username and password will be sent over the network/internet unencrypted, which means that they may be visible to others.
+    > For more information on how to configure your DataMiner System to use HTTPS, see [Setting up HTTPS on a DMA](xref:Setting_up_HTTPS_on_a_DMA).
+
+   - Other configuration settings can be ignored.
+
+1. Click *Save & test*. A *Success* message should be displayed.
+
+## Features
+
+### Generic Query Interface (GQI)
+
+A GQI query should be specified in JSON format. To get a correctly configured query, you can make use of the [DataMiner Dashboards app](xref:newR_D):
+
+1. In the Dashboards app, create a dashboard and create a query.
+
+   > [!TIP]
+   >
+   > - See [Creating and configuring dashboards](xref:Creating_and_configuring_dashboards).
+   > - See [Configuring query (GQI) data feeds](xref:Configuring_GQI_feeds)
+
+1. Visualize this query on the dashboard.
+
+1. Press F12 to open the developer tools and go to the *Network* tab.
+
+1. Refresh the dashboard and find the *OpenQuerySession* network call.
+
+1. Under payload, right-click *query:* and select *Copy value*.
+
+1. Open Grafana and paste the copied JSON code into Grafana.
+
+#### Time range filter
+
+Your GQI query can contain one or more *datetime* filters. To apply the *time range* filter from Grafana, you can manually edit the filter condition of the GQI query by replacing the numeric timestamp value with one of the following placeholders: `[gf-starttime]` and `[gf-endtime]`.
+
+Example:
+
+``` JSON
+{
+    "ID": "Value",
+    "Value": {
+        "Value": [gf-starttime],
+        "Type": "datetime",
+        "__type": "Skyline.DataMiner.Web.Common.v1.DMAPrimitiveValue"
+    },
+    "Type": "datetime",
+    "__type": "Skyline.DataMiner.Web.Common.v1.DMAGenericInterfaceQueryChosenOption"
+}
+```
+
+#### Transformations
+
+You may have to add a transformation to some GQI results. This could be the case when you have a GQI query that gets the latitude and longitude from element properties.
+
+Since properties in DataMiner are strings, you will need to add a *Convert field type* transformation to convert the latitude and longitude values to numbers before you can display them on a map visualization.
+
+#### Annotations
+
+You can use data from a GQI result to annotate the data on your dashboard. This can be configured in the settings of your Grafana dashboard.
+
+> [!TIP]
+> For more information about annotations, see [Annotate visualizations](https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/annotate-visualizations/).
+
+### Trend data
+
+To get the trend data of a parameter, specify the parameter using the following format: `dmaID/elementID/parameterID` or `dmaID/elementID/tableColumnParameterID/tableRowIndex`.
```

Comment:
Where do they need to specify this parameter?

Reply:
In Grafana when editing a panel, in the Query section using the DataMiner data source, choose for "Trend" and enter this parameter.